### DISCUSSION TITLE
Improve Documentation and Enable Continuous Previewing

### DISCUSSION THREAD

**Discussion by bassoy** [boostorg/ublas#139] [Ideas]
The improvement idea is to continuously create ascii-doctor documentation in order to
- validate the master documentation output
- preview the develop documentation output

**Comment by shikharvashistha** [comment]
I've been  researching on the continuous integration for the asciidoctor documentation and concluded few things those are:

1. We can build whole project and check for html preview or validate the html using these:
[first](https://github.com/marketplace/actions/html-preview) && [second](https://github.com/marketplace/actions/proof-html)
 
2. We can release a github page without using boost.build via:
[first](https://github.com/marketplace/actions/asciidoctor-ghpages) 

3. In place compile them without using the super project through this:
[first](https://github.com/Analog-inc/asciidoctor-action/)

4. html preview or validate the html file generated from 3rd step using these:
[first](https://github.com/marketplace/actions/html-preview) && [second](https://github.com/marketplace/actions/proof-html)

-  We can build whole project using boost build including all the html's (which we're not doing now "need to change the JAMFILE") and release a github page. I've created a sample [here](https://shikharvashistha.github.io/gh-pages-for-ublas-docs/doc/html/ublas.html) for reference.

**Reply by bassoy** [reply]
Comment is from @shikharvashistha:

I've been experimenting with the actions and successfully achieved :

I was able to compile asciidoc to html and create a auto release of github pages [here](https://github.com/shikharvashistha/ublas/deployments/activity_log?environment=github-pages) && [here](https://github.com/shikharvashistha/ublas/tree/docs).

Creating a html preview (only be generated when there is a change in compiled html file from asciidoc via actions in the commit) attaching example for reference [this](https://github.com/shikharvashistha/ublas/pull/5#issuecomment-938399498) && [this](https://github.com/shikharvashistha/ublas/tree/docs/.github/workflows).

**Reply by bassoy** [reply]
@shikharvashistha
I like your last approach and I think you should follow the last approach using Github Actions.

**Reply by shikharvashistha** [reply]
> @shikharvashistha I like your last approach and I think you should follow the last approach using Github Actions.

Yes I agree, to release a preview of documentation as discussed I'll be needing a dedicated branch which will look something like [this](https://github.com/shikharvashistha/ublas/tree/docs). Is it feasible to use `feature/doc` for the implementation of my approach ?

**Reply by bassoy** [reply]
Yes please use `feature/doc` and let's see how it looks like.

**Reply by bassoy** [reply]
I propose to create a PR for the develop branch with the following changes:
- Integrate relevant contents of [docs.yml](https://github.com/boostorg/ublas/blob/dd51ba9dbe1d10d769ccfd34268602a40f0c8538/.github/workflows/docs.yml) into [documentation.yml](https://github.com/boostorg/ublas/blob/dd51ba9dbe1d10d769ccfd34268602a40f0c8538/.github/workflows/documentation.yml)
- The specific action should be triggered for the develop branch
- The action should also contain a removal of the index.html on the remote before executing manoelcampos/asciidoctor-ghpages-action@v2

**Comment by shikharvashistha** [comment]
As there isn't a way to actually verify the correctness of documentation after some changes been done via commits in a pull request, so I would like to propose continuous previewing of documentation whenever any changes are being made to the same. Any suggestions are more than welcomed.

Wishes
@shikharvashistha
